### PR TITLE
Sorting out our `sort` icons

### PR DESCRIPTION
**Pull request by karsa-mistmere**
This pull request originates from #1004 by @danielbayley

During its review I noticed what I believed were inconsistencies about arrow directions in our current sort icons ("surely, sort-asc should point _downwards_"), after doing some research, I believe these perceived inconsistencies probably stem from cultural differences, a summary of what I've found:

Icon sets where `asc=up`, `desc=down`:
- Lucide
- [Octicons](https://primer.style/design/foundations/icons/sort-asc-24)
- [css.gg](https://css.gg/app) [basically, although implicitly]
- [Remix icon](https://remixicon.com/)

Icon sets where `asc=up`, `desc=down`:
- [Fontawesome v4](https://fontawesome.com/v4/icon/sort-alpha-asc) 
- [Phosphor Icons](https://phosphoricons.com/)
- [Tabler Icons](https://tablericons.com/)

Icon sets where the order is not stated, but instead only the direction is used (up or down):
- [Fontawesome v6](https://fontawesome.com/search?q=a-z&o=r) 
- [Bootstrap](https://icons.getbootstrap.com/?q=sort)
- [Unicons](https://iconscout.com/unicons/explore/line?redirect=%2Funicons%2Fexplore%2Fline%3Fcolor%3D%2525230092E4%26name%3Dsort-amount-down)
- [Zwicon](https://www.zwicon.com/cheatsheet.html)

---

In this PR I propose to follow in the footsteps of the "mainstream" and renamed our sort icons to not obviously reflect any order so that the exact usage is up to the end users (aliases were added for backwards compatibility):
e.g. `sort-asc` becomes `arrow-up-narrow-wide` and `sort-desc` becomes `arrow-down-wide-narrow`

Further icons supporting ascending and descending numerical and alphabetical order were also added:
image

As a side effect, the now incorrectly named `arrow-up-down` and `arrow-left-right` were made consistent with the sort icons and matching pairs were added (`arrow-down-up` and `arrow-right-left`).
(We may choose to retain the legacy design under a different name, as they have a significantly different feel).

### CODE REVIEW

**Comment by ericfennis** [comment]
@karsa-mistmere Nice work. Good that we take a look at the whole set of sort icons.
I see we use also the small arrows here. Can we try couple of icons with a larger arrow?

I think the larger arrow from the old one are a bit clearer viewed on 100% scale.
image

**Comment by karsa-mistmere** [comment]
Yeah, I think we can make that work.

1st row: originals
2nd & 3rd rows: current PR
4th & 5th rows: with bigger arrows (same head as current arrow-up-down and 2px longer than currently)
6th & 7th rows: with bigger arrows (same head as current arrow-up-down but same length as currently)

image

What do we think 4&5 vs 6&7?

**Comment by ericfennis** [comment]
@karsa-mistmere Nice work. To me the bottom row looks the most balanced, because the arrows is also aligning with the "A-Z" and "1-0" others are not. Also the `arrow-left-right` looks more cleaner.
I really want to see those icons on 100% scale (24x24px). Can you provide a figma link or something?

**Comment by karsa-mistmere** [comment]
Sure: https://www.figma.com/file/fXhFTLEiHKM4narKn35La9/lucide%2Fsort?node-id=0%3A1&t=q9dFXjl14pzBiD67-1

**Comment by ericfennis** [comment]
Thanks, looking at this I think we should go with the bottom row. Let me know what you think.

**Comment by karsa-mistmere** [comment]
> @karsa-mistmere Can you update the icons?

@ericfennis: Done! :)

**Review by ericfennis — APPROVED** [review]

File: icons/arrow-down-0-1.svg [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  stroke="currentColor"
+  stroke-width="2"
+  stroke-linecap="round"
+  stroke-linejoin="round"
+>
+  <path d="m3 16 4 4 4-4" />
+  <path d="M7 20V4" />
+  <path d="M17 10c-1.1 0-2-.9-2-2V6c0-1.1.9-2 2-2s2 .9 2 2v2c0 1.1-.9 2-2 2Z" />
+  <path d="M17 20v-6h-2" />
+  <path d="M15 20h4" />
```
Comment (karsa-mistmere) [inline icons/arrow-down-0-1.svg:16]:
```suggestion
  <rect x="15" y="4" width="4" height="6" ry="2" />
  <path d="m3 16 4 4 4-4" />
  <path d="M7 20V4" />
  <path d="M17 20v-6h-2" />
  <path d="M15 20h4" />
```

File: icons/arrow-down-1-0.svg [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  stroke="currentColor"
+  stroke-width="2"
+  stroke-linecap="round"
+  stroke-linejoin="round"
+>
+  <path d="m3 16 4 4 4-4" />
+  <path d="M7 20V4" />
+  <path d="M17 10V4h-2" />
+  <path d="M15 10h4" />
+  <path d="M17 20c-1.1 0-2-.9-2-2v-2c0-1.1.9-2 2-2s2 .9 2 2v2c0 1.1-.9 2-2 2Z" />
```
Comment (karsa-mistmere) [inline icons/arrow-down-1-0.svg:16]:
```suggestion

  <rect x="15" y="14" width="4" height="6" ry="2" />
  <path d="m3 16 4 4 4-4" />
  <path d="M7 20V4" />
  <path d="M17 10V4h-2" />
  <path d="M15 10h4" />
```

File: icons/arrow-up-0-1.svg [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  stroke="currentColor"
+  stroke-width="2"
+  stroke-linecap="round"
+  stroke-linejoin="round"
+>
+  <path d="m3 8 4-4 4 4" />
+  <path d="M7 4v16" />
+  <path d="M17 10c-1.1 0-2-.9-2-2V6c0-1.1.9-2 2-2s2 .9 2 2v2c0 1.1-.9 2-2 2Z" />
+  <path d="M17 20v-6h-2" />
+  <path d="M15 20h4" />
```
Comment (karsa-mistmere) [inline icons/arrow-up-0-1.svg:16]:
```suggestion
  <rect x="15" y="4" width="4" height="6" ry="2" />
  <path d="m3 8 4-4 4 4" />
  <path d="M7 4v16" />
  <path d="M17 20v-6h-2" />
  <path d="M15 20h4" />
```

File: icons/arrow-up-1-0.svg [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  stroke="currentColor"
+  stroke-width="2"
+  stroke-linecap="round"
+  stroke-linejoin="round"
+>
+  <path d="m3 8 4-4 4 4" />
+  <path d="M7 4v16" />
+  <path d="M17 10V4h-2" />
+  <path d="M15 10h4" />
+  <path d="M17 20c-1.1 0-2-.9-2-2v-2c0-1.1.9-2 2-2s2 .9 2 2v2c0 1.1-.9 2-2 2Z" />
```
Comment (karsa-mistmere) [inline icons/arrow-up-1-0.svg:16]:
```suggestion
  <rect x="15" y="14" width="4" height="6" ry="2" />
  <path d="m3 8 4-4 4 4" />
  <path d="M7 4v16" />
  <path d="M17 10V4h-2" />
  <path d="M15 10h4" />
```

**Comment by karsa-mistmere** [comment]
I let @jguddas's script do some optimization magic before merging. :)

**Comment by danielbayley** [comment]
> I let @jguddas's script do some optimization magic

@karsa-mistmere @jguddas Where might I take a look at this script? :)

**Comment by jguddas** [comment]
> > I let Jakob Guddas's script do some optimization magic
> 
> Karsa Jakob Guddas Where might I take a look at this script? :)

https://github.com/lucide-icons/lucide/blob/8a1e5581e6073be2ab03dbebd06bacedeb5a7ea4/site/src/pages/api/edit/optimize.ts

It's WIP but usable, you just have to press the optimize button 😄.

https://lucide-site-git-feat-added-edit-page-jguddas.vercel.app/edit/karsa-mistmere/lucide/arrow-down-0-1?PHN2ZwogIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIKICB3aWR0aD0iMjQiCiAgaGVpZ2h0PSIyNCIKICB2aWV3Qm94PSIwIDAgMjQgMjQiCiAgZmlsbD0ibm9uZSIKICBzdHJva2U9ImN1cnJlbnRDb2xvciIKICBzdHJva2Utd2lkdGg9IjIiCiAgc3Ryb2tlLWxpbmVjYXA9InJvdW5kIgogIHN0cm9rZS1saW5lam9pbj0icm91bmQiCj4KICA8cGF0aCBkPSJtNCAxNyAzIDMgMy0zIiAvPgogIDxwYXRoIGQ9Ik03IDIwVjQiIC8+CiAgPHBhdGggZD0iTTE3IDEwYTIuMCAyLjAwIDAgMCAxLTItMlY2YTIuMDAgMi4wMCAwIDAgMSA0IDB2MmEyLjAwIDIuMDAgMCAwIDEtMiAyIiAvPgogIDxwYXRoIGQ9Ik0xNyAyMHYtNmgtMiIgLz4KICA8cGF0aCBkPSJNMTUgMjBoNCIgLz4KPC9zdmc+

**Comment by karsa-mistmere** [comment]
#1022

**Comment by danielbayley** [comment]
> It's WIP but usable

Nice! @jguddas

Will this also be made part of `pnpm optimize`?

Could/should these optimisations exist as `svgo` plugins?

Also, just curious what the reason is for using `for i`[`n`] loops over `map`/`forEach`?

**Comment by jguddas** [comment]
> Will this also be made part of pnpm optimize?

Some parts are a bit flaky, so I don't know how well this can be integrated, but we could absolutely run this as a PR workflow.

> Could/should these optimisations exist as svgo plugins?

Yes, some are hard to generalize, but I could absolutely do that for most of them.

I don't know how well svgo is maintained tho.

> Also, just curious what the reason is for using for i[n] loops over map/forEach?

I have not spent any time cleaning things up, the code is all over the place.

**Comment by danielbayley** [comment]
> > Could/should these optimisations exist as svgo plugins?
> 
> I could absolutely do that for most of them.

> I don't know how well svgo is maintained tho.

Last commit was not that long ago. Besides, it supports [custom plugins](https://github.com/svg/svgo#custom-plugin)…

**Comment by danielbayley** [comment]
> > Could/should these optimisations exist as svgo plugins?
> 
> I could absolutely do that for most of them.

Cool. I had some SVGO plugins to write on my todo list…

Here are some current thoughts on optimisation, based on those, plus more I have come across in this project and your script…

* Remove `id`s/`class`es (always added by Adobe Illustrator)
* Merge overlapping anchor points
* Remove unnecessary anchor points
* Sort `width`, `height`, `x*`, `y*`, `r*` etc… attributes into something like that order
* Remove (e.g `rect`) `ry` attribute if `rx` is present (maybe this is necessary in some cases…?)
* Convert `line`s to `path`s (@jguddas should this be done in all cases?)

Am I right in thinking though, that all these cases are now covered, between the current builtin SVGO plugins, and this script?

> I don't know how well svgo is maintained tho.

Last commit was not that long ago. Also, it supports [custom plugins](https://github.com/svg/svgo#custom-plugin)…

**Pull request merged**